packages/react-ui-components/src/utils/useFontsApi.ts

```diff
@@ -0,0 +1,21 @@
+import { useCallback, useMemo } from 'react';
+
+const FONT_NOT_FOUND_ERROR = `Font was not found, make sure the font is included in your css`;
+
+export const useFontsApi = (fontFamily: string, fontSize: number) => {
+	const fontName = useMemo(() => `${fontSize}px ${fontFamily}`, [fontFamily, fontSize]);
+
+	const checkFont = useCallback(() => document.fonts.check(fontName), [fontName]);
+
+	const loadFont = useCallback(async () => {
+		if (checkFont()) return;
```

Comment:
What about a `isFontAvailable` memo instead of a callback? Seems better to me.
Also, add a return type to this hook.

```typescript
export const useFontsApi = (fontFamily: string, fontSize: number) => {
    const fontName = useMemo(() => `${fontSize}px ${fontFamily}`, [fontFamily, fontSize]);
    const isFontAvailable = useMemo(() => document.fonts.check(fontName), [fontName]);

	const checkFont = useCallback(() => document.fonts.check(fontName), [fontName]);
	const loadFont = useCallback(async () => {
		if (isFontAvailable) return;

		try {
			await document.fonts.load(fontName);
		} catch (error) {
			throw new Error(FONT_NOT_FOUND_ERROR);
		}
	}, [fontName]);

	return { checkFont, loadFont, isFontAvailable };
};
```

Reply:
looks good to me :D